src/data/equiv/basic.lean

```diff
@@ -1906,6 +1906,15 @@ begin
   rw swap_apply_of_ne_of_ne hi hj,
 end
 
+lemma swap_apply_eq_iff [decidable_eq α] {x y z w : α} :
+  swap x y z = w ↔ z = swap x y w :=
+begin
+  split;
```

Comment:
A simpler proof probably starts `rw equiv.apply_eq_iff_eq_symm_apply`, and finished with some other obvious rewrite